### PR TITLE
Added script, to help gather requests made to aws when running terraform

### PR DESCRIPTION
**Pull request by detnon**
Added helper script and some information in the Readme regarding usage

### CODE REVIEW

File: cybersecuritytools/aws_permissions_helper/aws_requests.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import re
+import sys
+
+pattern = "((?<=DEBUG: Request\s)(\w*\W\w*))"
+
+with open(sys.argv[1],"r") as data:
+    with open('aws_requests.txt','w') as aws:
+        for s in sorted(set([re.search(pattern,d).group(1) for d in data if re.search(pattern, d)])):
```
Comment (0atman) [inline cybersecuritytools/aws_permissions_helper/aws_requests.py]: m8

Comment (0atman) [reply cybersecuritytools/aws_permissions_helper/aws_requests.py]: This is very clever, a bit too clever for me to understand. Would you rename to be clearer or give names to the following:
```
pattern
sys.argv[1]
data
aws
s
d
``` 

The problem here is nothing is named, or the names add very little value.

**Review by 0atman — CHANGES_REQUESTED** [review]
first thing's first :ok_hand:

**Comment by akinnane** [comment]
Can you either make the whole script a one-liner or break it out into one-simple-thing-per-line?

File: cybersecuritytools/aws_permissions_helper/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Aws Permissions Helper 
+
+This script will read the a logfile and extract all the requests made to create the resources and output them to a file called `aws_requests.txt`
+
+## Usage
+
+Firstly, you will need to enable the correct logging level by terraform, before the next step run:
+
+```bash
+export TF_LOG=TRACE
```
Comment (danjoneslf) [inline cybersecuritytools/aws_permissions_helper/README.md:10]: Presumably if your pattern is looking for DEBUG log lines you can set `export TF_LOG=DEBUG` and make the log file smaller?

File: cybersecuritytools/aws_permissions_helper/aws_requests.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import re
+import sys
+
+pattern = "((?<=DEBUG: Request\s)(\w*\W\w*))"
+
+with open(sys.argv[1],"r") as data:
+    with open('aws_requests.txt','w') as aws:
```
Comment (danjoneslf) [inline cybersecuritytools/aws_permissions_helper/aws_requests.py]: Deniz showed me a think where you create the 2 contexts and use a single `with` like 

```
terraform_log = open(sys.argv[1],"r") 
aws_request_log = open('aws_requests.txt','w')

with terraform_log, aws_requests_log: 
    ....
```

something like that anyway.

File: cybersecuritytools/aws_permissions_helper/aws_requests.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    with open(input, "r") as raw_log:
+        debug_log = raw_log.readlines()
+
+    requests = extract_requests(debug_log)
+
+    with open("aws_requests.json", "w") as aws_requests:
+        aws_requests.write(grouped_permissions(requests))
+
+
+def first_match(pattern: Pattern[str], text: str) -> str:
```
Comment (akinnane) [inline cybersecuritytools/aws_permissions_helper/aws_requests.py:22]: Is this right? Should it really return "No Match" when there isn't a match?

File: cybersecuritytools/aws_permissions_helper/aws_requests.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    """
+    grouped_permissions: Dict[str, List[str]] = {}
+    for request in requests.splitlines():
+        match_service = first_match(SERVICE, request)
+        match_permission = first_match(PERMISSION, request)
+        if match_service not in grouped_permissions:
+            grouped_permissions[match_service] = []
+            grouped_permissions[match_service].append(match_permission)
+        else:
+            grouped_permissions[match_service].append(match_permission)
```
Comment (akinnane) [inline cybersecuritytools/aws_permissions_helper/aws_requests.py]:
```suggestion
        grouped_permissions.setdefault(match_service, []).append(match_permission)
```

You can use the `setdefault` method on dict's to either get and return the object if the key is set, or save a default object into the key and return that if the key is not set. That's a more elegant way of expressing the above if block. 

https://docs.python.org/3/library/stdtypes.html#dict

**Review by akinnane — APPROVED** [review]

File: cybersecuritytools/aws_permissions_helper/aws_requests.py (hunk 5 above)
Comment (detnon) [reply cybersecuritytools/aws_permissions_helper/aws_requests.py]: Nice! Updated to use this

File: cybersecuritytools/aws_permissions_helper/aws_requests.py (hunk 4 above)
Comment (detnon) [reply cybersecuritytools/aws_permissions_helper/aws_requests.py:22]: I wasn't sure what to return here, any suggestions?

Its highly unlikely that anything at this point will not match but probably should have a good way to handle that

**Review by 0atman — APPROVED** [review]
assuming all the existing comments are ok, sure!

**Pull request merged**